src/java/org/apache/cassandra/service/consensus/migration/ConsensusMigrationMutationHelper.java

```diff
@@ -275,21 +270,27 @@ public SplitMutation(@Nullable T accordMutation, @Nullable T untrackedMutation,
 
     public <T extends IMutation> SplitMutation<T> splitMutation(T mutation, ClusterMetadata cm)
     {
-        boolean isTracked = isTrackedMutation(mutation);
+        Token token = mutation.key().getToken();
+        Predicate<TableId> isTrackedUpdate = tableId -> MigrationRouter.shouldUseTrackedForWrites(cm, mutation.getKeyspaceName(), tableId, token);
+        Predicate<TableId> isUntrackedUpdate = not(isTrackedUpdate);
         if (mutation.potentialTxnConflicts().allowed)
-            return new SplitMutation<>(null, isTracked ? null : mutation, isTracked ? mutation : null);
+        {
```

Comment:
This check for potentialTxnConflicts is odd. Looks like I added it, but it barely makes sense and is misleading because I don't think you would ever split mutations from a context where txn conflicts are allowed?

That flag is set on mutations from Paxos and Accord when the mutation is for commit and it tells the checks in `ColumnFamilyStore` to not generate an error if there could be a potential transaction conflict.

Maybe remove this and add an illegal state exception if transaction conflicts are allowed with a message saying it's not needed/supported.